website/docs/mutation.md

```diff
@@ -0,0 +1,278 @@
+
+Note that the resource is not filtered if an element is not present or an empty list.
+
+#### Intent
+
+This specifies what should be changed in the resource.
+
+An example of the section is shown below:
+```yaml
+    location: "spec.containers[name:foo].imagePullPolicy"
+    parameters:
+       assign:
+        value: "Always"
+```
+
+The `location` element specifies the path to be modified.
+The `parameters.assign.value` element specifies the value to be set for the element specified in `location`. Note that the value can either be a simple string or a composite value.
+
+An example of a composite value:
+```yaml
+    location: "spec.containers[name:networking]"
+    parameters: 
+      assign: 
+        value: 
+          name: "networking"
+          imagePullPolicy: Always
+
+```
+
+The `location` element can specify either a simple subelement or an element in a list.
+For example the location `spec.containers[name:foo].imagePullPolicy` would be parsed as follows:
+- ***spec**.containers[name:foo].imagePullPolicy* - the spec element
+- *spec.**containers[name:foo]**.imagePullPolicy* - container subelement of spec. The container element is a list. Out of the list choosen, an element with the `name` element having the value `foo`.
+ - *spec.containers[name:foo].**imagePullPolicy*** - in the element from the list chosen in the previous step the element `imagePullPolicy` is chosen
+
+The yaml illustrating the above `location`:
+```yaml
+spec:
+   containers:
+   -  name:foo
+      imagePullPolicy:
+```
+Note that when the list element is empty (`spec.containers[].imagePullPolicy`), all the list element will be updated.
+
+##### Conditionals
+
+The conditions for updating the resource.
+Two types of conditions exist:
+- path tests - a resource will only be updated when a specified path exists or not
+- value tests - a resource will only be updated when the existing value is/is not contained in a list of values
+
+An example of the conditionals: 
+```yaml
+parameters:
+     pathTests:
+          - subPath: "spec.containers[name:foo]"
+            condition: MustExist
+          - subPath: spec.containers[name:foo].securityContext.capabilities
+            condition: MustNotExist
+
+      ifIn: [<lists of current values that cause the value to be modified>]
+      ifNotIn: [<lists of current values which will cause the value NOT to be modified>]
+
+```
+
+
+### AssignMetadata
+
+The AssignMetadata is a CRD for modifying the metadata section of a resource. Note that the metadata of a resource is a very sensitive piece of data, and certain mutations could result in unintended consequences. An example of this could be changing the name or namespace of a resource. The AssignMetadata changes have therefore been limited to only the labels and annotations. Furthermore, it is currently only allowed to add a label or annotation, but not modify an existing one (this might change in later versions of the feature).
```

Comment:
Do we want to highlight that this might change? I don't know if we have a plausible path forward for this yet, though discussion is active and ongoing.

We don't need a fully-realized design, but we should have something that people agree is more likely to work than not.

Reply:
Removed